langserver/symbol.go

```diff
@@ -488,3 +479,91 @@ var symbolCacheTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
 func init() {
 	prometheus.MustRegister(symbolCacheTotal)
 }
+
+func listPkgs(ctxt *build.Context, dir string) []string {
+	ch := make(chan string)
```

Comment:
I would expect it would potentially make perf worse! The thing consuming this channel is a very tight loop appending to a slice. So will be like a buffered channel, without the need of extra ceremony that buffered channel likely needs to do.